File: C Sharp/C# Advanced/HW3_NumeralSystem/Task4_HexToDecimal/HexToDecimal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Task4_HexToDecimal
{
    class HexToDecimal
    {
        static void Main()
        {
            string hexNumber = Console.ReadLine();
            long decimalNumber = ConvertHexToDecimal(hexNumber);
            Console.WriteLine(decimalNumber);
        }

        private static long ConvertHexToDecimal(string hexNumber)
        {
            long sum = 0;
            for (int i = hexNumber.Length - 1; i >= 0; i--)
            {
                
                if (hexNumber[i] >= '0' && hexNumber[i] <= '9')
                {
                    sum += (long)Math.Pow(16, (hexNumber.Length - i - 1)) * (hexNumber[i] - '0');
                }
                else
                {
                    sum += (long)Math.Pow(16, (hexNumber.Length - i - 1)) * ((hexNumber[i] - 'A') + 10);
                }
            }
            return sum;
        }
    }
}

File: Front-end/JavaScript-Fundamentals/04. Loops/hexToDecimal.js
function solve(args) {
    let decimalNumber,
        currentNum,
        sum = 0;
    str = args + '';
    for (let i = str.length - 1; i >= 0; i -= 1) {
        currentNum = +str[i];
        if (currentNum >= 0 && currentNum <= 9) {
            decimalNumber = Math.pow(16, (str.length - i - 1)) * (currentNum);
            sum = sum + decimalNumber;
        } else {
            decimalNumber = Math.pow(16, (str.length - i - 1)) * ((str.charCodeAt(i) - 65) + 10);
            sum = sum + decimalNumber;
        }
    }
    console.log(sum);
}


//solve(['4ED528CBB4']);